src/OrchardCoreContrib.PoExtractor.Razor/MetadataProviders/RazorMetadataProvider.cs

```diff
@@ -38,7 +36,17 @@ public string GetContext(SyntaxNode node)
 
             var path = node.SyntaxTree.FilePath.TrimStart(_basePath);
 
-            return path.Replace(Path.DirectorySeparatorChar, '.').Replace(_razorExtension, string.Empty);
+            path = RemoveFileExtension(path);
+            
+            return path.Replace(Path.DirectorySeparatorChar, '.');
+        }
+
+        private static string RemoveFileExtension(string path)
+        {
+            var lastPeriod = path.LastIndexOf('.');
+            return lastPeriod < 0 ?
+                path : // No extension was found
+                path.Substring(0, lastPeriod);
```

Comment:
```csharp
            var path = new StringBuilder(node.SyntaxTree.FilePath.TrimStart(_basePath));
            path.Replace(Path.DirectorySeparatorChar, '.');

            if (path.ToString().Contains(_razorPageExtension))
            {
                path.Replace(_razorPageExtension, string.Empty);
            }
            else
            {
                path.Replace(_razorComponentExtension, string.Empty);
            }

            return path.ToString();
```

Reply:
a few notes:
-the `Contains(_razorPageExtension)` means we search for the `_razorPageExtension` twice in case it exists (both `.Contains(_razorPageExtension)` and `Replace(_razorPageExtension)` search for the same string). I don't think that's necessary.
- having `.Replace(Path.DirectorySeparatorChar, '.')' before the other replaces means that if we have a path like `Feature/razor/pages/SomePage/razor` is converted 1st to "Feature.razor.pages.SomePage.razor" and then to "Feature.pages.SomePage", that is the "razor" folder name is lost - that seems buggy. And it can easily be resolved by putting the replace of the `Path.DirectorySeparatorChar` at the end.
- there should be no need to complicate this by using `StringBuilder`, since it's a small string and few replacements are done - and furthermore there's no evidence that `StringBuilder` performs better in such a case, which I think would be needed to be sure, see for example [here](https://medium.com/@nuno.caneco/dont-believe-everything-you-see-at-stackoverflow-28a043336ca8)